### DISCUSSION TITLE
Why socket.broadcast.emit()  broadcast to every sockets including the sender?

### DISCUSSION THREAD

**Discussion by HOLYYUN** [socketio/socket.io#3999] [Q&A]
code
 `socket.broadcast.emit('someone create ',{roomList:roomList});`
and all the clients in namespace "/hall"
the emit had   broadcast to everyone,including the sender.

**Comment by darrachequesne** [comment]
Hi! I'm not able to reproduce:

Server:

```js
const io = require("socket.io")(3000);

io.of("/test").on("connection", socket => {
  socket.broadcast.emit("hello");
});
```

Client:

```js
const socket = require("socket.io-client")("http://localhost:3000/test");

socket.on("hello", () => {
  console.log("got hello");
});
```

Which version are you using?

**Reply by HOLYYUN** [reply]
I'm using v4.0
Server:

```
const io=require("socket.io")(3000);
 const hall = io.of("/hall“);

 hall.on("connection",socket => {
     socket.leave(socket.id); //leave the default room beacause of the operational requirements
    socket.on("create room",(roomName)=>{
    if(roomName){
    socket.broadcast.emit("create success");//to broadcast every sockets but the sender someone created a room success;
    socket.to(socket.id).emit("create done");// emit the sender created done;
    //the two emit do not work in my way,broadcast.emit() to every sockets  including the sender and  socket.to(socket.id).emit() not worked;
}
})
})
```
Client:

```
const socket = require("socket.io-client")("http://localhost:3000/");
   //some fuction
   socket.emit("create room",roomName);
   
  socket.on("create success",()=>{
   //do something
});
socket.on("create done",()=>{
    //do something
})
```
Now I guess the problem produced because of the client socket require without the namespace "/hall"?

**Reply by darrachequesne** [reply]
Thanks for providing a way to reproduce. I think this is due to `socket.leave(socket.id);`, because it messes with the way sockets are excluded from the broadcast.

Out of curiosity, could you please explain what your operational requirements are?

**Reply by HOLYYUN** [reply]
when the client connected，we need to get all the rooms in hall,so need to leave the default room because  the room is not build by us.
Thank you for the help!
